### PR TITLE
Add tab for UCES scenarios

### PR DESCRIPTION
**Pull request by karst**
The order of the scenarios is listed in the same order listed on the UCES website. This also allows for translating the scenarios to other languages!

If you don't have any of the UCES scenarios installed the game will look like this:
image

If you do have UCES installed:
image

### CODE REVIEW

**Comment by IntelOrca** [comment]
Hardcoding this doesn't scale when you want to add more and more custom scenario packs. While these changes could eventually be replaced with something more flexible (.parkpack or something), the plugin API changes and scenario category would have to stay. So I would avoid adding those extra plugin strings, because it would be better if you could define a pack with its own categories. And source game would be other, as you would have a field for the scenario pack it belongs to instead.

**Comment by Gymnasiast** [comment]
I once started some rudimentary work on adding something like that. I should perhaps try and revive that.

**Comment by Gymnasiast** [comment]
As IntelOrca points out, the plug-in stuff would have to stay in the game, even when it’s spun off. But perhaps you can remove that and have plugins simply return "Other" for these kinds of scenarios (and maybe they already do, you updated the .ts file but didn’t add the lower case strings anywhere else). It seems to me that that is what IntelOrca wants in the first place.

> It does give the desired result for now. 

Sure, but you can say that about lots of things, down to the dirtiest hack or kludge. We have to look at the longer term too, especially since we have to deal with any consequences.

Nevertheless, I think we can merge this if you take out the plug-in stuff. We will need a way to check if the plug-in API returns the correct results, though - perhaps someone can write a simple one that just displays this info in a box.

**Comment by IntelOrca** [comment]
> Currently with all existing hardcoded tabs it scales correctly.

By scale I meant software scalability. It requires too much code and maintenance for each scenario pack added.

**Comment by Gymnasiast** [comment]
> > Currently with all existing hardcoded tabs it scales correctly.
> 
> By scale I meant software scalability. It requires too much code and maintenance for each scenario pack added.

Would you say we should hold this one off because of that? (Assuming the plug-in stuff gets taken out)

**Comment by karst** [comment]
Ah i seem to have misunderstood. I have based the PR of mimicking and learning as I go. Mimicking mostly so I don't forget things. I did not realize some of the code i touch can be used permanently by outside sources. Ill remove that. I do think UCES is worthy of hardcoding for now myself as it's the most important non-official scenario pack. I would love to see more scenario packs but I think we should wait with that for people to have access to scenariopack objects.

**Comment by karst** [comment]
> If the plugin stuff gets removed, I don't really mind since it can just get reverted. The other thing to note is that new scenarios should not be able to select those new categories.

I'll look into this

**Comment by Gymnasiast** [comment]
@IntelOrca I added code to the scripting UI to properly fall back to "Other" if it cannot map the scenario category or source game. This is the kind of data the scripting UI now puts out:
```
{ path: '/home/michael/Programma's/RollerCoaster Tycoon 2/drive_c/Spelletjes/RollerCoaster Tycoon 2/Scenarios/Cleveland's Luna Park.SC6',
  id: 255,
  category: 'other',
  sourceGame: 'other',
  internalName: 'Cleveland's Luna Park',
  name: 'Luna Park, Cleveland',
  details: 'As it was, Opening Day -- 18 MAY 1905.{NEWLINE}Author: Aetherwave',
  highscore: 
   { name: '',
     companyValue: 3592470 } }
{ path: '/home/michael/Programma's/RollerCoaster Tycoon 2/drive_c/Spelletjes/RollerCoaster Tycoon 2/Scenarios/Mount Vesuvius 1700 A.SC6',
  id: 255,
  category: 'other',
  sourceGame: 'other',
  internalName: 'Mount Vesuvius 1700 A.D. by Katatude for UCES',
  name: 'Mount Vesuvius',
  details: 'Pompeii and Herculaneum were burried by Mt. Vesuvius in 79 A.D. Visit the excavations and build a park!{NEWLINE}Author: Katatude',
  highscore: null }
  ```

**Comment by Legotown55** [comment]
How do i get the new tabs in the game

**Comment by Gymnasiast** [comment]
> How do i get the new tabs in the game

Wait until the PR is merged, then install the latest develop.

**Review by IntelOrca — APPROVED** [review]

File: distribution/changelog.txt
Comment (Gymnasiast) [inline distribution/changelog.txt:16]:
```suggestion
- Improved: [#18607] A new tab for all UCES Scenarios, if it’s installed.
```
Yes, I am _that_ anal.

File: data/language/en-GB.txt
Comment (Gymnasiast) [inline data/language/en-GB.txt:4474]:
```suggestion
STR_DTLS    :The Romans are tired of boring gladiator fights. Give them a better thrill, turn a Roman city into the greatest amusement park of all time!{NEWLINE}Author: Kaffe
```

File: data/language/en-GB.txt
Comment (Gymnasiast) [inline data/language/en-GB.txt:4483]:
```suggestion
STR_DTLS    :Mum built this scene under our tree. Now she wants a park made in it! Can you help?{NEWLINE}Author: Katatude
```

(Change back for en-US)

File: data/language/en-GB.txt
Comment (Gymnasiast) [inline data/language/en-GB.txt:4459]:
```suggestion
STR_DTLS    :The Time Machine. Build to go - when you want, where you want. Eternity awaits. It’s all relative.{NEWLINE}Author: Katatude
```

File: data/language/en-GB.txt
Comment (Gymnasiast) [inline data/language/en-GB.txt:4454]:
```suggestion
STR_DTLS    :A rockslide damaged your railway. Your workers have gone prospecting. Is there gold in roller coasters?{NEWLINE}Authors: Squid, Buckone, Fossil
```
Change back for en-US.

File: data/language/en-GB.txt
Comment (Gymnasiast) [inline data/language/en-GB.txt:4418]:
```suggestion
###
<UCES Halloween>
```

File: data/language/en-GB.txt
Comment (tupaschoal) [inline data/language/en-GB.txt:4514]: "attract more quests"
Should this have read "guests"?

File: data/language/en-GB.txt
Comment (tupaschoal) [inline data/language/en-GB.txt:4508]: Should this have dropped the 'II'?

File: data/language/en-GB.txt
Comment (Gymnasiast) [reply data/language/en-GB.txt:4514]: It should have, yes.

**Pull request merged**
